src/mongo_logger.ts

```diff
@@ -760,15 +764,26 @@ export class MongoLogger {
     this.maxDocumentLength = options.maxDocumentLength;
     this.logDestination = options.logDestination;
     this.logDestinationIsStdErr = options.logDestinationIsStdErr;
+    this.createWillLog();
   }
 
-  willLog(severity: SeverityLevel, component: MongoLoggableComponent): boolean {
-    return compareSeverity(severity, this.componentSeverities[component]) <= 0;
+  createWillLog() {
+    for (const component of Object.values(MongoLoggableComponent)) {
+      this.willLog[component] = Object.fromEntries(
+        Object.entries(SeverityLevel).map(([_key, value]) => [
+          value as SeverityLevel,
+          compareSeverity(value, this.componentSeverities[component]) <= 0
+        ])
+      ) as Record<SeverityLevel, boolean>;
+    }
```

Comment:
(nit): I am to sure if this is easier to read, but at least we can avoid casting.

```js
createWillLog() {
  for (const component of Object.values(MongoLoggableComponent)) {
    for (const severityLevel of Object.values(SeverityLevel)) {
      this.willLog[component] = {
        ...this.willLog[component],
        [severityLevel]: compareSeverity(severityLevel, this.componentSeverities[component]) <= 0
      };
    }
  }
}
```

Maybe there is even a better way to refactor it. Feel free to explore or ignore if this is a common way it is done elsewhere in the codebase.

Reply:
I see. One worry I have about this way to refactor it is having to run the spread `...` complexity for each entry in `willLog`. Do you have any suggestions on how to avoid casting and avoid using spread?

Reply:
Would `Object.assign()` work?

```js
  for (const component of Object.values(MongoLoggableComponent)) {
    for (const severityLevel of Object.values(SeverityLevel)) {
      this.willLog[component] = Object.assign({}, this.willLog[component], {
        [severityLevel]: compareSeverity(severityLevel, this.componentSeverities[component]) <= 0
      });
    }
  }
}
```

Reply:
I believe we're trying to use `Object.assign()` less in the Node Driver and it has the same O(N) complexity as spread. I've found a typecasting work around though (in my latest commit).